src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -326,21 +326,25 @@ public bool IsSet(AvaloniaProperty property)
 
             VerifyAccess();
 
-            var description = GetDescription(source);
-
             if (property.IsDirect)
             {
                 if (property.IsReadOnly)
                 {
                     throw new ArgumentException($"The property {property.Name} is readonly.");
                 }
 
-                Logger.Verbose(
-                    LogArea.Property, 
-                    this,
-                    "Bound {Property} to {Binding} with priority LocalValue", 
-                    property, 
-                    description);
+                if (Logger.IsEnabled(LogEventLevel.Verbose))
```

Comment:
We should check for both log level and log area, I think.

Reply:
I think the better API would be a `TryGetLogger` method that returns `null` when logging is not configured, so the logging call would look like

```cs
Logger.TryGetLogger(LogEventLevel.Verbose, LogArea.Property)
    ?.Log(this, "Bound {Property} to {Binding} with priority LocalValue", property,  GetDescription(source));

```
which is more clear than `if` block and passing the log level twice.

Reply:
There is no way of checking if area is enabled it only decorates log output.
And I agree `TryGetLogger` API is nicer, will change mine to use something like that.

Reply:
Added new API as well.